### PR TITLE
Implemented: Added the support for using facility selector from dxp-component(#dxp/228)

### PR DESCRIPTION
**Pull request by R-Sourabh**
### Related Issues

https://github.com/hotwax/dxp-components/issues/228

### Short Description and Why It's Useful

- Added support for using the `facility selector` on the settings page from the `dxp-component`.
- Replaced the app-level facility state and updated the logic to use the dxp-component for facility `state` management.
- Removed code related to storing/fetching facilities and currentFacility in the application.
- Utilized `getters` from the user store in dxp-components across the entire application.

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/fulfillment#contribution-guideline)

### CODE REVIEW

File: src/store/modules/stock/actions.ts [hunk 1]
```diff
@@ -6,11 +6,13 @@ import * as types from './mutation-types'
 import { hasError } from '@/adapter'
 import logger from '@/logger'
 import { showToast } from '@/utils'
-import { translate } from '@hotwax/dxp-components'
+import { translate, useUserStore } from '@hotwax/dxp-components'
 
 const actions: ActionTree<StockState, RootState> = {
   async fetchStock({ commit }, { productId, facilityId = '' }) {
-    const id = facilityId ? facilityId : this.state.user.currentFacility.facilityId
+    const getCurrentFacility: any = useUserStore().getCurrentFacility
```
Comment (ravilodhi) [inline src/store/modules/stock/actions.ts]: Use getCurrentFacilityId() util function here.

File: src/store/modules/stock/getters.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
 import { GetterTree } from 'vuex'
 import StockState from './StockState'
 import RootState from '../../RootState'
+import { useUserStore } from "@hotwax/dxp-components";
 
 const getters: GetterTree <StockState, RootState> = {
-  getProductStock: (state, getters, RootState) => (productId: any, facilityId?: any) => {
-    const id = facilityId ? facilityId : RootState.user.currentFacility.facilityId
+  getProductStock: (state, getters) => (productId: any, facilityId?: any) => {
```
Comment (ravilodhi) [inline src/store/modules/stock/getters.ts:7]: Use the getCurrentFacilityId() util function here.

File: src/store/modules/transferorder/actions.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ const actions: ActionTree<TransferOrderState, RootState> = {
     emitter.emit('presentLoader');
     let resp;
     const transferOrderQuery = JSON.parse(JSON.stringify(state.transferOrder.query))
+    const currentFacilityId = getCurrentFacilityId();
```
Comment (ravilodhi) [inline src/store/modules/transferorder/actions.ts]: Use the getCurrentFacilityId() directly if currentFacilityId variable is not reused.

File: src/views/Settings.vue [hunk 4]
```diff
@@ -38,6 +38,7 @@
 
       <section>
         <DxpOmsInstanceNavigator />
+        <DxpFacilitySwitcher @updateFacility="handleFacilityUpdate()"/>
```
Comment (dt2patel) [inline src/views/Settings.vue]: this is the wrong place, facility selector should be after the product store selector. @ravilodhi

**Review by dt2patel — COMMENTED** [review]
Incorrect location of facility switcher

File: src/views/Settings.vue [hunk 5]
```diff
@@ -38,6 +38,7 @@
 
       <section>
         <DxpOmsInstanceNavigator />
+        <DxpFacilitySwitcher @updateFacility="updateFacility()"/>
```
Comment (dt2patel) [inline src/views/Settings.vue:41]: Incorrect sequence, should be after product store selector

**Pull request merged**